Add --reference argument for cram decompression

ASLeonard opened the pull request:
This was helpful for non-standard references or if there were any issues with firewalls like

```
[E::easy_errno] Libcurl reported error 60 (SSL peer certificate or SSH remote key was not OK)
[W::find_file_url] Failed to open reference "https://www.ebi.ac.uk/ena/cram/md5/9f1e20f56671ea323b75f6a3356048bb": Input/output error
```

wdecoster commented:
Hi, thanks for the suggestion and changes! As you can see, the tests are not passing at the moment, and it looks like this is the case because you didn't edit the test functions. I would use an `Option<String>` as the type for this argument, or is there a reason you implemented it as is?

ASLeonard commented:
>  is there a reason you implemented it as is?

Learning rust on the fly, so plenty of room for improvement 😃.

 I've changed it to `Option<String>` and modified the test to match the function signature. Ultimately probably should add an explicit test for using the `--reference` arg, but at least this is backward compatible.

wdecoster commented:
Thanks!
